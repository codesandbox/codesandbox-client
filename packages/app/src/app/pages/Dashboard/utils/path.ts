export function getParentPath(path: string) {
  if (path === '/' || path === '') {
    return '/';
  }

  const split = path.split('/');
  split.pop();

  return split.join('/');
}

export function normalizePath(path: string) {
  if (!path.startsWith('/')) {
    return '/' + path;
  }
  return path;
}
