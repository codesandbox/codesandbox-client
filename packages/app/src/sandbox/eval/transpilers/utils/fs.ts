import type { Module } from 'sandpack-core/lib/types/module';
import { ChildHandler } from '../worker-transpiler/child-handler';

export async function getModulesFromMainThread(opts: {
  childHandler: ChildHandler;
  loaderContextId: number;
}): Promise<Module[]> {
  const { childHandler, loaderContextId } = opts;
  if (loaderContextId == null) {
    throw new Error('Loader context id is required');
  }

  const { modules } = await childHandler.callFn({
    method: 'resolve-fs',
    data: {
      loaderContextId,
    },
  });
  return modules;
}
