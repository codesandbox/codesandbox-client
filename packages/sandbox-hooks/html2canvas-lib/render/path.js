export var PathType;
(function (PathType) {
    PathType[PathType["VECTOR"] = 0] = "VECTOR";
    PathType[PathType["BEZIER_CURVE"] = 1] = "BEZIER_CURVE";
})(PathType || (PathType = {}));
export const equalPath = (a, b) => {
    if (a.length === b.length) {
        return a.some((v, i) => v === b[i]);
    }
    return false;
};
export const transformPath = (path, deltaX, deltaY, deltaW, deltaH) => {
    return path.map((point, index) => {
        switch (index) {
            case 0:
                return point.add(deltaX, deltaY);
            case 1:
                return point.add(deltaX + deltaW, deltaY);
            case 2:
                return point.add(deltaX + deltaW, deltaY + deltaH);
            case 3:
                return point.add(deltaX, deltaY + deltaH);
        }
        return point;
    });
};
