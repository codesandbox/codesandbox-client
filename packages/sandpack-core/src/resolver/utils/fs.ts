import { Gensync } from 'gensync';
import { EMPTY_SHIM } from './constants';

export type FnIsFile = Gensync<(filepath: string) => boolean>;
export type FnReadFile = Gensync<(filepath: string) => string>;

export function* isFile(
  filepath: string,
  isFileFn: FnIsFile
): Generator<any, boolean, any> {
  if (filepath === EMPTY_SHIM) {
    return true;
  }
  return yield* isFileFn(filepath);
}

export function getParentDirectories(
  filepath: string,
  rootDir: string = '/'
): string[] {
  const parts = filepath.split('/');
  const directories = [];
  while (parts.length > 0) {
    const directory = parts.join('/') || '/';
    // Test /foo vs /foo-something - /foo-something is not in rootDir
    if (directory.length < rootDir.length || !directory.startsWith(rootDir)) {
      break;
    }
    directories.push(directory);
    parts.pop();
  }
  return directories;
}
