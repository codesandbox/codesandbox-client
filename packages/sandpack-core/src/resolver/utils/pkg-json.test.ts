import fs from 'fs';
import path from 'path';

import { processPackageJSON } from './pkg-json';

const FIXTURE_PATH = path.join(__dirname, '../fixture');

describe('process package.json', () => {
  it('Should correctly process pkg.exports from @babel/runtime', () => {
    const content = JSON.parse(
      fs.readFileSync(
        path.join(FIXTURE_PATH, 'node_modules/@babel/runtime/package.json'),
        'utf-8'
      )
    );
    const processedPkg = processPackageJSON(
      content,
      '/node_modules/@babel/runtime'
    );
    expect(processedPkg).toMatchSnapshot();
  });

  it('Should correctly handle nested pkg#exports fields (solid-js)', () => {
    const content = JSON.parse(
      fs.readFileSync(
        path.join(FIXTURE_PATH, 'node_modules/solid-js/package.json'),
        'utf-8'
      )
    );
    const processedPkg = processPackageJSON(content, '/node_modules/solid-js');
    expect(processedPkg).toMatchSnapshot();
  });

  it('Should correctly handle root pkg.json', () => {
    const content = JSON.parse(
      fs.readFileSync(path.join(FIXTURE_PATH, 'package.json'), 'utf-8')
    );
    const processedPkg = processPackageJSON(content, '/');
    expect(processedPkg).toMatchSnapshot();
  });
});
