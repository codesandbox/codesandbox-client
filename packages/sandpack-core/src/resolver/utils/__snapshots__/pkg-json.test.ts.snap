// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`process package.json Should correctly handle nested pkg#exports fields (solid-js) 1`] = `
Object {
  "aliases": Object {
    "/node_modules/solid-js": "/node_modules/solid-js/dist/solid.cjs",
    "/node_modules/solid-js/dist/*": "/node_modules/solid-js/dist/$1",
    "/node_modules/solid-js/dist/server.cjs": "/node_modules/solid-js/dist/solid.cjs",
    "/node_modules/solid-js/dist/server.js": "/node_modules/solid-js/dist/solid.js",
    "/node_modules/solid-js/h": "/node_modules/solid-js/h/dist/h.cjs",
    "/node_modules/solid-js/h/dist/*": "/node_modules/solid-js/h/dist/$1",
    "/node_modules/solid-js/html": "/node_modules/solid-js/html/dist/html.cjs",
    "/node_modules/solid-js/html/dist/*": "/node_modules/solid-js/html/dist/$1",
    "/node_modules/solid-js/store": "/node_modules/solid-js/store/dist/store.cjs",
    "/node_modules/solid-js/store/dist/*": "/node_modules/solid-js/store/dist/$1",
    "/node_modules/solid-js/universal": "/node_modules/solid-js/universal/dist/dev.cjs",
    "/node_modules/solid-js/universal/dist/*": "/node_modules/solid-js/universal/dist/$1",
    "/node_modules/solid-js/web": "/node_modules/solid-js/web/dist/web.cjs",
    "/node_modules/solid-js/web/dist/*": "/node_modules/solid-js/web/dist/$1",
  },
  "hasExports": true,
}
`;

exports[`process package.json Should correctly handle root pkg.json 1`] = `
Object {
  "aliases": Object {
    "aliased": "foo",
    "aliased-file": "/bar.js",
    "aliased-file/*": "/bar.js/$1",
    "aliased/*": "foo/$1",
    "aliasedabsolute": "/nested",
    "aliasedabsolute/*": "/nested/$1",
    "aliasedfolder": "/nested",
    "aliasedfolder/*": "/nested/$1",
    "foo/bar": "/bar.js",
    "foo/bar/*": "/bar.js/$1",
    "glob/*/*": "/nested/$2",
    "glob/*/*/*": "/nested/$2/$1",
    "something": "/nested/test.js",
    "something/*": "/nested/test.js/$1",
  },
  "hasExports": false,
}
`;

exports[`process package.json Should correctly process pkg.exports from @babel/runtime 1`] = `
Object {
  "aliases": Object {
    "/node_modules/@babel/runtime/helpers/AsyncGenerator": "/node_modules/@babel/runtime/helpers/AsyncGenerator.js",
    "/node_modules/@babel/runtime/helpers/AwaitValue": "/node_modules/@babel/runtime/helpers/AwaitValue.js",
    "/node_modules/@babel/runtime/helpers/applyDecoratedDescriptor": "/node_modules/@babel/runtime/helpers/applyDecoratedDescriptor.js",
    "/node_modules/@babel/runtime/helpers/arrayLikeToArray": "/node_modules/@babel/runtime/helpers/arrayLikeToArray.js",
    "/node_modules/@babel/runtime/helpers/arrayWithHoles": "/node_modules/@babel/runtime/helpers/arrayWithHoles.js",
    "/node_modules/@babel/runtime/helpers/arrayWithoutHoles": "/node_modules/@babel/runtime/helpers/arrayWithoutHoles.js",
    "/node_modules/@babel/runtime/helpers/assertThisInitialized": "/node_modules/@babel/runtime/helpers/assertThisInitialized.js",
    "/node_modules/@babel/runtime/helpers/asyncGeneratorDelegate": "/node_modules/@babel/runtime/helpers/asyncGeneratorDelegate.js",
    "/node_modules/@babel/runtime/helpers/asyncIterator": "/node_modules/@babel/runtime/helpers/asyncIterator.js",
    "/node_modules/@babel/runtime/helpers/asyncToGenerator": "/node_modules/@babel/runtime/helpers/asyncToGenerator.js",
    "/node_modules/@babel/runtime/helpers/awaitAsyncGenerator": "/node_modules/@babel/runtime/helpers/awaitAsyncGenerator.js",
    "/node_modules/@babel/runtime/helpers/classApplyDescriptorDestructureSet": "/node_modules/@babel/runtime/helpers/classApplyDescriptorDestructureSet.js",
    "/node_modules/@babel/runtime/helpers/classApplyDescriptorGet": "/node_modules/@babel/runtime/helpers/classApplyDescriptorGet.js",
    "/node_modules/@babel/runtime/helpers/classApplyDescriptorSet": "/node_modules/@babel/runtime/helpers/classApplyDescriptorSet.js",
    "/node_modules/@babel/runtime/helpers/classCallCheck": "/node_modules/@babel/runtime/helpers/classCallCheck.js",
    "/node_modules/@babel/runtime/helpers/classCheckPrivateStaticAccess": "/node_modules/@babel/runtime/helpers/classCheckPrivateStaticAccess.js",
    "/node_modules/@babel/runtime/helpers/classCheckPrivateStaticFieldDescriptor": "/node_modules/@babel/runtime/helpers/classCheckPrivateStaticFieldDescriptor.js",
    "/node_modules/@babel/runtime/helpers/classExtractFieldDescriptor": "/node_modules/@babel/runtime/helpers/classExtractFieldDescriptor.js",
    "/node_modules/@babel/runtime/helpers/classNameTDZError": "/node_modules/@babel/runtime/helpers/classNameTDZError.js",
    "/node_modules/@babel/runtime/helpers/classPrivateFieldDestructureSet": "/node_modules/@babel/runtime/helpers/classPrivateFieldDestructureSet.js",
    "/node_modules/@babel/runtime/helpers/classPrivateFieldGet": "/node_modules/@babel/runtime/helpers/classPrivateFieldGet.js",
    "/node_modules/@babel/runtime/helpers/classPrivateFieldLooseBase": "/node_modules/@babel/runtime/helpers/classPrivateFieldLooseBase.js",
    "/node_modules/@babel/runtime/helpers/classPrivateFieldLooseKey": "/node_modules/@babel/runtime/helpers/classPrivateFieldLooseKey.js",
    "/node_modules/@babel/runtime/helpers/classPrivateFieldSet": "/node_modules/@babel/runtime/helpers/classPrivateFieldSet.js",
    "/node_modules/@babel/runtime/helpers/classPrivateMethodGet": "/node_modules/@babel/runtime/helpers/classPrivateMethodGet.js",
    "/node_modules/@babel/runtime/helpers/classPrivateMethodSet": "/node_modules/@babel/runtime/helpers/classPrivateMethodSet.js",
    "/node_modules/@babel/runtime/helpers/classStaticPrivateFieldDestructureSet": "/node_modules/@babel/runtime/helpers/classStaticPrivateFieldDestructureSet.js",
    "/node_modules/@babel/runtime/helpers/classStaticPrivateFieldSpecGet": "/node_modules/@babel/runtime/helpers/classStaticPrivateFieldSpecGet.js",
    "/node_modules/@babel/runtime/helpers/classStaticPrivateFieldSpecSet": "/node_modules/@babel/runtime/helpers/classStaticPrivateFieldSpecSet.js",
    "/node_modules/@babel/runtime/helpers/classStaticPrivateMethodGet": "/node_modules/@babel/runtime/helpers/classStaticPrivateMethodGet.js",
    "/node_modules/@babel/runtime/helpers/classStaticPrivateMethodSet": "/node_modules/@babel/runtime/helpers/classStaticPrivateMethodSet.js",
    "/node_modules/@babel/runtime/helpers/construct": "/node_modules/@babel/runtime/helpers/construct.js",
    "/node_modules/@babel/runtime/helpers/createClass": "/node_modules/@babel/runtime/helpers/createClass.js",
    "/node_modules/@babel/runtime/helpers/createForOfIteratorHelper": "/node_modules/@babel/runtime/helpers/createForOfIteratorHelper.js",
    "/node_modules/@babel/runtime/helpers/createForOfIteratorHelperLoose": "/node_modules/@babel/runtime/helpers/createForOfIteratorHelperLoose.js",
    "/node_modules/@babel/runtime/helpers/createSuper": "/node_modules/@babel/runtime/helpers/createSuper.js",
    "/node_modules/@babel/runtime/helpers/decorate": "/node_modules/@babel/runtime/helpers/decorate.js",
    "/node_modules/@babel/runtime/helpers/defaults": "/node_modules/@babel/runtime/helpers/defaults.js",
    "/node_modules/@babel/runtime/helpers/defineEnumerableProperties": "/node_modules/@babel/runtime/helpers/defineEnumerableProperties.js",
    "/node_modules/@babel/runtime/helpers/defineProperty": "/node_modules/@babel/runtime/helpers/defineProperty.js",
    "/node_modules/@babel/runtime/helpers/esm/AsyncGenerator": "/node_modules/@babel/runtime/helpers/esm/AsyncGenerator.js",
    "/node_modules/@babel/runtime/helpers/esm/AwaitValue": "/node_modules/@babel/runtime/helpers/esm/AwaitValue.js",
    "/node_modules/@babel/runtime/helpers/esm/applyDecoratedDescriptor": "/node_modules/@babel/runtime/helpers/esm/applyDecoratedDescriptor.js",
    "/node_modules/@babel/runtime/helpers/esm/arrayLikeToArray": "/node_modules/@babel/runtime/helpers/esm/arrayLikeToArray.js",
    "/node_modules/@babel/runtime/helpers/esm/arrayWithHoles": "/node_modules/@babel/runtime/helpers/esm/arrayWithHoles.js",
    "/node_modules/@babel/runtime/helpers/esm/arrayWithoutHoles": "/node_modules/@babel/runtime/helpers/esm/arrayWithoutHoles.js",
    "/node_modules/@babel/runtime/helpers/esm/assertThisInitialized": "/node_modules/@babel/runtime/helpers/esm/assertThisInitialized.js",
    "/node_modules/@babel/runtime/helpers/esm/asyncGeneratorDelegate": "/node_modules/@babel/runtime/helpers/esm/asyncGeneratorDelegate.js",
    "/node_modules/@babel/runtime/helpers/esm/asyncIterator": "/node_modules/@babel/runtime/helpers/esm/asyncIterator.js",
    "/node_modules/@babel/runtime/helpers/esm/asyncToGenerator": "/node_modules/@babel/runtime/helpers/esm/asyncToGenerator.js",
    "/node_modules/@babel/runtime/helpers/esm/awaitAsyncGenerator": "/node_modules/@babel/runtime/helpers/esm/awaitAsyncGenerator.js",
    "/node_modules/@babel/runtime/helpers/esm/classApplyDescriptorDestructureSet": "/node_modules/@babel/runtime/helpers/esm/classApplyDescriptorDestructureSet.js",
    "/node_modules/@babel/runtime/helpers/esm/classApplyDescriptorGet": "/node_modules/@babel/runtime/helpers/esm/classApplyDescriptorGet.js",
    "/node_modules/@babel/runtime/helpers/esm/classApplyDescriptorSet": "/node_modules/@babel/runtime/helpers/esm/classApplyDescriptorSet.js",
    "/node_modules/@babel/runtime/helpers/esm/classCallCheck": "/node_modules/@babel/runtime/helpers/esm/classCallCheck.js",
    "/node_modules/@babel/runtime/helpers/esm/classCheckPrivateStaticAccess": "/node_modules/@babel/runtime/helpers/esm/classCheckPrivateStaticAccess.js",
    "/node_modules/@babel/runtime/helpers/esm/classCheckPrivateStaticFieldDescriptor": "/node_modules/@babel/runtime/helpers/esm/classCheckPrivateStaticFieldDescriptor.js",
    "/node_modules/@babel/runtime/helpers/esm/classExtractFieldDescriptor": "/node_modules/@babel/runtime/helpers/esm/classExtractFieldDescriptor.js",
    "/node_modules/@babel/runtime/helpers/esm/classNameTDZError": "/node_modules/@babel/runtime/helpers/esm/classNameTDZError.js",
    "/node_modules/@babel/runtime/helpers/esm/classPrivateFieldDestructureSet": "/node_modules/@babel/runtime/helpers/esm/classPrivateFieldDestructureSet.js",
    "/node_modules/@babel/runtime/helpers/esm/classPrivateFieldGet": "/node_modules/@babel/runtime/helpers/esm/classPrivateFieldGet.js",
    "/node_modules/@babel/runtime/helpers/esm/classPrivateFieldLooseBase": "/node_modules/@babel/runtime/helpers/esm/classPrivateFieldLooseBase.js",
    "/node_modules/@babel/runtime/helpers/esm/classPrivateFieldLooseKey": "/node_modules/@babel/runtime/helpers/esm/classPrivateFieldLooseKey.js",
    "/node_modules/@babel/runtime/helpers/esm/classPrivateFieldSet": "/node_modules/@babel/runtime/helpers/esm/classPrivateFieldSet.js",
    "/node_modules/@babel/runtime/helpers/esm/classPrivateMethodGet": "/node_modules/@babel/runtime/helpers/esm/classPrivateMethodGet.js",
    "/node_modules/@babel/runtime/helpers/esm/classPrivateMethodSet": "/node_modules/@babel/runtime/helpers/esm/classPrivateMethodSet.js",
    "/node_modules/@babel/runtime/helpers/esm/classStaticPrivateFieldDestructureSet": "/node_modules/@babel/runtime/helpers/esm/classStaticPrivateFieldDestructureSet.js",
    "/node_modules/@babel/runtime/helpers/esm/classStaticPrivateFieldSpecGet": "/node_modules/@babel/runtime/helpers/esm/classStaticPrivateFieldSpecGet.js",
    "/node_modules/@babel/runtime/helpers/esm/classStaticPrivateFieldSpecSet": "/node_modules/@babel/runtime/helpers/esm/classStaticPrivateFieldSpecSet.js",
    "/node_modules/@babel/runtime/helpers/esm/classStaticPrivateMethodGet": "/node_modules/@babel/runtime/helpers/esm/classStaticPrivateMethodGet.js",
    "/node_modules/@babel/runtime/helpers/esm/classStaticPrivateMethodSet": "/node_modules/@babel/runtime/helpers/esm/classStaticPrivateMethodSet.js",
    "/node_modules/@babel/runtime/helpers/esm/construct": "/node_modules/@babel/runtime/helpers/esm/construct.js",
    "/node_modules/@babel/runtime/helpers/esm/createClass": "/node_modules/@babel/runtime/helpers/esm/createClass.js",
    "/node_modules/@babel/runtime/helpers/esm/createForOfIteratorHelper": "/node_modules/@babel/runtime/helpers/esm/createForOfIteratorHelper.js",
    "/node_modules/@babel/runtime/helpers/esm/createForOfIteratorHelperLoose": "/node_modules/@babel/runtime/helpers/esm/createForOfIteratorHelperLoose.js",
    "/node_modules/@babel/runtime/helpers/esm/createSuper": "/node_modules/@babel/runtime/helpers/esm/createSuper.js",
    "/node_modules/@babel/runtime/helpers/esm/decorate": "/node_modules/@babel/runtime/helpers/esm/decorate.js",
    "/node_modules/@babel/runtime/helpers/esm/defaults": "/node_modules/@babel/runtime/helpers/esm/defaults.js",
    "/node_modules/@babel/runtime/helpers/esm/defineEnumerableProperties": "/node_modules/@babel/runtime/helpers/esm/defineEnumerableProperties.js",
    "/node_modules/@babel/runtime/helpers/esm/defineProperty": "/node_modules/@babel/runtime/helpers/esm/defineProperty.js",
    "/node_modules/@babel/runtime/helpers/esm/extends": "/node_modules/@babel/runtime/helpers/esm/extends.js",
    "/node_modules/@babel/runtime/helpers/esm/get": "/node_modules/@babel/runtime/helpers/esm/get.js",
    "/node_modules/@babel/runtime/helpers/esm/getPrototypeOf": "/node_modules/@babel/runtime/helpers/esm/getPrototypeOf.js",
    "/node_modules/@babel/runtime/helpers/esm/inherits": "/node_modules/@babel/runtime/helpers/esm/inherits.js",
    "/node_modules/@babel/runtime/helpers/esm/inheritsLoose": "/node_modules/@babel/runtime/helpers/esm/inheritsLoose.js",
    "/node_modules/@babel/runtime/helpers/esm/initializerDefineProperty": "/node_modules/@babel/runtime/helpers/esm/initializerDefineProperty.js",
    "/node_modules/@babel/runtime/helpers/esm/initializerWarningHelper": "/node_modules/@babel/runtime/helpers/esm/initializerWarningHelper.js",
    "/node_modules/@babel/runtime/helpers/esm/instanceof": "/node_modules/@babel/runtime/helpers/esm/instanceof.js",
    "/node_modules/@babel/runtime/helpers/esm/interopRequireDefault": "/node_modules/@babel/runtime/helpers/esm/interopRequireDefault.js",
    "/node_modules/@babel/runtime/helpers/esm/interopRequireWildcard": "/node_modules/@babel/runtime/helpers/esm/interopRequireWildcard.js",
    "/node_modules/@babel/runtime/helpers/esm/isNativeFunction": "/node_modules/@babel/runtime/helpers/esm/isNativeFunction.js",
    "/node_modules/@babel/runtime/helpers/esm/isNativeReflectConstruct": "/node_modules/@babel/runtime/helpers/esm/isNativeReflectConstruct.js",
    "/node_modules/@babel/runtime/helpers/esm/iterableToArray": "/node_modules/@babel/runtime/helpers/esm/iterableToArray.js",
    "/node_modules/@babel/runtime/helpers/esm/iterableToArrayLimit": "/node_modules/@babel/runtime/helpers/esm/iterableToArrayLimit.js",
    "/node_modules/@babel/runtime/helpers/esm/iterableToArrayLimitLoose": "/node_modules/@babel/runtime/helpers/esm/iterableToArrayLimitLoose.js",
    "/node_modules/@babel/runtime/helpers/esm/jsx": "/node_modules/@babel/runtime/helpers/esm/jsx.js",
    "/node_modules/@babel/runtime/helpers/esm/maybeArrayLike": "/node_modules/@babel/runtime/helpers/esm/maybeArrayLike.js",
    "/node_modules/@babel/runtime/helpers/esm/newArrowCheck": "/node_modules/@babel/runtime/helpers/esm/newArrowCheck.js",
    "/node_modules/@babel/runtime/helpers/esm/nonIterableRest": "/node_modules/@babel/runtime/helpers/esm/nonIterableRest.js",
    "/node_modules/@babel/runtime/helpers/esm/nonIterableSpread": "/node_modules/@babel/runtime/helpers/esm/nonIterableSpread.js",
    "/node_modules/@babel/runtime/helpers/esm/objectDestructuringEmpty": "/node_modules/@babel/runtime/helpers/esm/objectDestructuringEmpty.js",
    "/node_modules/@babel/runtime/helpers/esm/objectSpread": "/node_modules/@babel/runtime/helpers/esm/objectSpread.js",
    "/node_modules/@babel/runtime/helpers/esm/objectSpread2": "/node_modules/@babel/runtime/helpers/esm/objectSpread2.js",
    "/node_modules/@babel/runtime/helpers/esm/objectWithoutProperties": "/node_modules/@babel/runtime/helpers/esm/objectWithoutProperties.js",
    "/node_modules/@babel/runtime/helpers/esm/objectWithoutPropertiesLoose": "/node_modules/@babel/runtime/helpers/esm/objectWithoutPropertiesLoose.js",
    "/node_modules/@babel/runtime/helpers/esm/possibleConstructorReturn": "/node_modules/@babel/runtime/helpers/esm/possibleConstructorReturn.js",
    "/node_modules/@babel/runtime/helpers/esm/readOnlyError": "/node_modules/@babel/runtime/helpers/esm/readOnlyError.js",
    "/node_modules/@babel/runtime/helpers/esm/set": "/node_modules/@babel/runtime/helpers/esm/set.js",
    "/node_modules/@babel/runtime/helpers/esm/setPrototypeOf": "/node_modules/@babel/runtime/helpers/esm/setPrototypeOf.js",
    "/node_modules/@babel/runtime/helpers/esm/skipFirstGeneratorNext": "/node_modules/@babel/runtime/helpers/esm/skipFirstGeneratorNext.js",
    "/node_modules/@babel/runtime/helpers/esm/slicedToArray": "/node_modules/@babel/runtime/helpers/esm/slicedToArray.js",
    "/node_modules/@babel/runtime/helpers/esm/slicedToArrayLoose": "/node_modules/@babel/runtime/helpers/esm/slicedToArrayLoose.js",
    "/node_modules/@babel/runtime/helpers/esm/superPropBase": "/node_modules/@babel/runtime/helpers/esm/superPropBase.js",
    "/node_modules/@babel/runtime/helpers/esm/taggedTemplateLiteral": "/node_modules/@babel/runtime/helpers/esm/taggedTemplateLiteral.js",
    "/node_modules/@babel/runtime/helpers/esm/taggedTemplateLiteralLoose": "/node_modules/@babel/runtime/helpers/esm/taggedTemplateLiteralLoose.js",
    "/node_modules/@babel/runtime/helpers/esm/tdz": "/node_modules/@babel/runtime/helpers/esm/tdz.js",
    "/node_modules/@babel/runtime/helpers/esm/temporalRef": "/node_modules/@babel/runtime/helpers/esm/temporalRef.js",
    "/node_modules/@babel/runtime/helpers/esm/temporalUndefined": "/node_modules/@babel/runtime/helpers/esm/temporalUndefined.js",
    "/node_modules/@babel/runtime/helpers/esm/toArray": "/node_modules/@babel/runtime/helpers/esm/toArray.js",
    "/node_modules/@babel/runtime/helpers/esm/toConsumableArray": "/node_modules/@babel/runtime/helpers/esm/toConsumableArray.js",
    "/node_modules/@babel/runtime/helpers/esm/toPrimitive": "/node_modules/@babel/runtime/helpers/esm/toPrimitive.js",
    "/node_modules/@babel/runtime/helpers/esm/toPropertyKey": "/node_modules/@babel/runtime/helpers/esm/toPropertyKey.js",
    "/node_modules/@babel/runtime/helpers/esm/typeof": "/node_modules/@babel/runtime/helpers/esm/typeof.js",
    "/node_modules/@babel/runtime/helpers/esm/unsupportedIterableToArray": "/node_modules/@babel/runtime/helpers/esm/unsupportedIterableToArray.js",
    "/node_modules/@babel/runtime/helpers/esm/wrapAsyncGenerator": "/node_modules/@babel/runtime/helpers/esm/wrapAsyncGenerator.js",
    "/node_modules/@babel/runtime/helpers/esm/wrapNativeSuper": "/node_modules/@babel/runtime/helpers/esm/wrapNativeSuper.js",
    "/node_modules/@babel/runtime/helpers/esm/wrapRegExp": "/node_modules/@babel/runtime/helpers/esm/wrapRegExp.js",
    "/node_modules/@babel/runtime/helpers/esm/writeOnlyError": "/node_modules/@babel/runtime/helpers/esm/writeOnlyError.js",
    "/node_modules/@babel/runtime/helpers/extends": "/node_modules/@babel/runtime/helpers/extends.js",
    "/node_modules/@babel/runtime/helpers/get": "/node_modules/@babel/runtime/helpers/get.js",
    "/node_modules/@babel/runtime/helpers/getPrototypeOf": "/node_modules/@babel/runtime/helpers/getPrototypeOf.js",
    "/node_modules/@babel/runtime/helpers/inherits": "/node_modules/@babel/runtime/helpers/inherits.js",
    "/node_modules/@babel/runtime/helpers/inheritsLoose": "/node_modules/@babel/runtime/helpers/inheritsLoose.js",
    "/node_modules/@babel/runtime/helpers/initializerDefineProperty": "/node_modules/@babel/runtime/helpers/initializerDefineProperty.js",
    "/node_modules/@babel/runtime/helpers/initializerWarningHelper": "/node_modules/@babel/runtime/helpers/initializerWarningHelper.js",
    "/node_modules/@babel/runtime/helpers/instanceof": "/node_modules/@babel/runtime/helpers/instanceof.js",
    "/node_modules/@babel/runtime/helpers/interopRequireDefault": "/node_modules/@babel/runtime/helpers/interopRequireDefault.js",
    "/node_modules/@babel/runtime/helpers/interopRequireWildcard": "/node_modules/@babel/runtime/helpers/interopRequireWildcard.js",
    "/node_modules/@babel/runtime/helpers/isNativeFunction": "/node_modules/@babel/runtime/helpers/isNativeFunction.js",
    "/node_modules/@babel/runtime/helpers/isNativeReflectConstruct": "/node_modules/@babel/runtime/helpers/isNativeReflectConstruct.js",
    "/node_modules/@babel/runtime/helpers/iterableToArray": "/node_modules/@babel/runtime/helpers/iterableToArray.js",
    "/node_modules/@babel/runtime/helpers/iterableToArrayLimit": "/node_modules/@babel/runtime/helpers/iterableToArrayLimit.js",
    "/node_modules/@babel/runtime/helpers/iterableToArrayLimitLoose": "/node_modules/@babel/runtime/helpers/iterableToArrayLimitLoose.js",
    "/node_modules/@babel/runtime/helpers/jsx": "/node_modules/@babel/runtime/helpers/jsx.js",
    "/node_modules/@babel/runtime/helpers/maybeArrayLike": "/node_modules/@babel/runtime/helpers/maybeArrayLike.js",
    "/node_modules/@babel/runtime/helpers/newArrowCheck": "/node_modules/@babel/runtime/helpers/newArrowCheck.js",
    "/node_modules/@babel/runtime/helpers/nonIterableRest": "/node_modules/@babel/runtime/helpers/nonIterableRest.js",
    "/node_modules/@babel/runtime/helpers/nonIterableSpread": "/node_modules/@babel/runtime/helpers/nonIterableSpread.js",
    "/node_modules/@babel/runtime/helpers/objectDestructuringEmpty": "/node_modules/@babel/runtime/helpers/objectDestructuringEmpty.js",
    "/node_modules/@babel/runtime/helpers/objectSpread": "/node_modules/@babel/runtime/helpers/objectSpread.js",
    "/node_modules/@babel/runtime/helpers/objectSpread2": "/node_modules/@babel/runtime/helpers/objectSpread2.js",
    "/node_modules/@babel/runtime/helpers/objectWithoutProperties": "/node_modules/@babel/runtime/helpers/objectWithoutProperties.js",
    "/node_modules/@babel/runtime/helpers/objectWithoutPropertiesLoose": "/node_modules/@babel/runtime/helpers/objectWithoutPropertiesLoose.js",
    "/node_modules/@babel/runtime/helpers/possibleConstructorReturn": "/node_modules/@babel/runtime/helpers/possibleConstructorReturn.js",
    "/node_modules/@babel/runtime/helpers/readOnlyError": "/node_modules/@babel/runtime/helpers/readOnlyError.js",
    "/node_modules/@babel/runtime/helpers/set": "/node_modules/@babel/runtime/helpers/set.js",
    "/node_modules/@babel/runtime/helpers/setPrototypeOf": "/node_modules/@babel/runtime/helpers/setPrototypeOf.js",
    "/node_modules/@babel/runtime/helpers/skipFirstGeneratorNext": "/node_modules/@babel/runtime/helpers/skipFirstGeneratorNext.js",
    "/node_modules/@babel/runtime/helpers/slicedToArray": "/node_modules/@babel/runtime/helpers/slicedToArray.js",
    "/node_modules/@babel/runtime/helpers/slicedToArrayLoose": "/node_modules/@babel/runtime/helpers/slicedToArrayLoose.js",
    "/node_modules/@babel/runtime/helpers/superPropBase": "/node_modules/@babel/runtime/helpers/superPropBase.js",
    "/node_modules/@babel/runtime/helpers/taggedTemplateLiteral": "/node_modules/@babel/runtime/helpers/taggedTemplateLiteral.js",
    "/node_modules/@babel/runtime/helpers/taggedTemplateLiteralLoose": "/node_modules/@babel/runtime/helpers/taggedTemplateLiteralLoose.js",
    "/node_modules/@babel/runtime/helpers/tdz": "/node_modules/@babel/runtime/helpers/tdz.js",
    "/node_modules/@babel/runtime/helpers/temporalRef": "/node_modules/@babel/runtime/helpers/temporalRef.js",
    "/node_modules/@babel/runtime/helpers/temporalUndefined": "/node_modules/@babel/runtime/helpers/temporalUndefined.js",
    "/node_modules/@babel/runtime/helpers/toArray": "/node_modules/@babel/runtime/helpers/toArray.js",
    "/node_modules/@babel/runtime/helpers/toConsumableArray": "/node_modules/@babel/runtime/helpers/toConsumableArray.js",
    "/node_modules/@babel/runtime/helpers/toPrimitive": "/node_modules/@babel/runtime/helpers/toPrimitive.js",
    "/node_modules/@babel/runtime/helpers/toPropertyKey": "/node_modules/@babel/runtime/helpers/toPropertyKey.js",
    "/node_modules/@babel/runtime/helpers/typeof": "/node_modules/@babel/runtime/helpers/typeof.js",
    "/node_modules/@babel/runtime/helpers/unsupportedIterableToArray": "/node_modules/@babel/runtime/helpers/unsupportedIterableToArray.js",
    "/node_modules/@babel/runtime/helpers/wrapAsyncGenerator": "/node_modules/@babel/runtime/helpers/wrapAsyncGenerator.js",
    "/node_modules/@babel/runtime/helpers/wrapNativeSuper": "/node_modules/@babel/runtime/helpers/wrapNativeSuper.js",
    "/node_modules/@babel/runtime/helpers/wrapRegExp": "/node_modules/@babel/runtime/helpers/wrapRegExp.js",
    "/node_modules/@babel/runtime/helpers/writeOnlyError": "/node_modules/@babel/runtime/helpers/writeOnlyError.js",
    "/node_modules/@babel/runtime/package": "/node_modules/@babel/runtime/package.json",
    "/node_modules/@babel/runtime/package.json": "/node_modules/@babel/runtime/package.json",
    "/node_modules/@babel/runtime/regenerator": "/node_modules/@babel/runtime/regenerator/index.js",
    "/node_modules/@babel/runtime/regenerator/": "/node_modules/@babel/runtime/regenerator/",
    "/node_modules/@babel/runtime/regenerator/*.js": "/node_modules/@babel/runtime/regenerator/$1.js",
  },
  "hasExports": true,
}
`;
